kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UIServicesClientImpl.java

```diff
@@ -107,18 +107,34 @@ public String getTaskForm(String containerId, Long taskId, String language) {
         return getTaskFormByType(containerId, taskId, language, ANY_FORM);
     }
 
+    @Override
+    public String getTaskFormAsUser(String userId, String containerId, Long taskId, String language) {
+        return getTaskFormByTypeAsUser(userId, containerId, taskId, language, ANY_FORM);
+    }
+
     @Override
     public String getTaskFormByType(String containerId, Long taskId, String language, String formType) {
-        return getTaskFormByType( containerId, taskId, language, formType, true );
+        return getTaskFormByTypeAsUser("", containerId, taskId, language, formType, true );
     }
 
-    private String getTaskFormByType( String containerId, Long taskId, String language, String formType, boolean marshallContent ) {
+    @Override
+    public String getTaskFormByTypeAsUser(String userId, String containerId, Long taskId, String language, String formType) {
+        return getTaskFormByTypeAsUser(userId, containerId, taskId, language, formType, true );
+    }
+
+    private String getTaskFormByTypeAsUser(String userId, String containerId, Long taskId, String language, String formType, boolean marshallContent ) {
         if( config.isRest() ) {
-            Map<String, Object> valuesMap = new HashMap<String, Object>();
+            Map<String, Object> valuesMap = new HashMap<>();
             valuesMap.put(RestURI.CONTAINER_ID, containerId);
             valuesMap.put(RestURI.TASK_INSTANCE_ID, taskId);
 
-            StringBuffer params = new StringBuffer();
+            String userQuery = userId.equals("") ? "" : getUserQueryStr(userId);
```

Comment:
This code can throw NPE when null userId is supplied. 
It would be good to replace userId default value in this class from "" to null, here you can check null value using StringUtils.isEmpty .

Reply:
Will do that too.